pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -342,7 +345,14 @@ private void sendOutputMessage(Record srcRecord, Object output) {
             Thread.currentThread().setContextClassLoader(functionClassLoader);
         }
         try {
-            this.sink.write(new SinkRecord<>(srcRecord, output));
+            SinkRecord sinkRecord;
```

Comment:
I don't understand why do you need this change here. What is the value for adding a SinkKVRecord? SinkRecord is a wrapper of the output object. The output object can be a KeyValue object.

Reply:
@sijie 
these are the key points:
1) on the Source side you use KVRecord for records with a KeyValue payload, this lets you specify the keySchema and the valueSchema, for consistency it is good to have the same situation on the Sink side
2) KeyValueSchema is not part of the public api (it is in the impl package), so in theory you cannot get the keySchema and the valueSchema when you are inside a Sink, if we implement KVRecord then you have those schemas
3) When we have AutoConsumeSchema that returns a KeyValue object the record.getSchema() is the internal class AutoConsumeSchema, so you cannot still get the keySchema and the valueSchema  (this problem is still not present, because we have not merged the Sink<GenericObject patch yet

Reply:
@sijie the alternative is:
- make KeyValueSchema part of the "public" API (or add an interface on the "api" package)
- hide "AutoConsumeSchema" in the Sink side and make Record#getSchema() return the current KeyValueSchema

but this alternative is not good because:
- we are going to add a new API to represent KeyValueSchema, but we already have KVRecord
- we are changing the behaviour of the Sink, not returning AutoConsumeSchema anymore